extensions/plugin-history-sync/src/queue.ts

```diff
@@ -4,27 +4,20 @@ import type { History } from "history";
  * This function is required to avoid any race conditions caused by asynchronous history updates.
  */
 export const makeQueue = (history: History) => {
-  let pending = false;
+  let previousTask = Promise.resolve();
 
   const queue = (cb: () => void) => {
```

Comment:
```suggestion
  const enqueue = (cb: () => void) => {
```

I thought queue -> enqueue might be better, so I'll give you a suggestion.

Reply:
Thanks for your suggestion and I came up with another idea:

How about `enqueueHistoryTask` as a name of the function so that can represent the more specific role of itself, even further it can help users notice that the task should contain history action at least more than one or the queue should results in a deadlock.

Reply:
I agree with the name `enqueueHistoryTask` because it is more detailed 😄

Reply:
Let's make compromises:

- `makeQueue`: rename to `makeHistoryTaskQueue` and return object with `queue` property rather than directly returning `queue` function.
- `queue`: rename to `enqueue`